### ISSUE TITLE
get_toc and set_toc do not behave consistently for rotated pages

### ISSUE THREAD

**Issue by blenzi** [pymupdf/PyMuPDF#2634]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
When calling `get_toc(simple=False)`, the coordinates of the point correspond to the **rotated** page while `set_toc` seems to expect **unrotated** coordinates. Therefore, calling `new.set_toc(pdf.get_toc(simple=False))` produces wrong toc links.

## To Reproduce (mandatory)

```
import fitz

with fitz.open("landscape_rot.pdf") as pdf, fitz.open() as new:
    new.insert_pdf(pdf)
    new.set_toc(pdf.get_toc(simple=False))
    
    for item in pdf.get_toc(simple=False): print(item)
    print()
    for item in new.get_toc(simple=False): print(item)
```

Gives:
```
[1, 'CHAPITRE I : PREAMBULE', 6, {'kind': 4, 'xref': 156, 'name': 'page=6&zoom=100,0,-49', 'collapse': False, 'zoom': 0.0}]
[2, '1 CONTEXTE POLITIQUE ET ENGAGEMENTS', 7, {'kind': 1, 'xref': 158, 'page': 6, 'to': Point(77.6199, 63.828), 'zoom': 0.0, 'collapse': True}]
[3, '1.1 CONTEXTE POLITIQUE INTERNATIONAL', 7, {'kind': 1, 'xref': 170, 'page': 6, 'to': Point(86.2342, 181.485), 'zoom': 0.0}]
[3, '1.2 CONTEXTE POLITIQUE EUROPEEN', 7, {'kind': 1, 'xref': 173, 'page': 6, 'to': Point(86.2342, 408.405), 'zoom': 0.0}]
[3, '1.3 CONTEXTE POLITIQUE FRANÇAIS', 7, {'kind': 1, 'xref': 171, 'page': 6, 'to': Point(86.2342, 559.905), 'zoom': 0.0}]
[2, '2 ETAT DES LIEUX', 7, {'kind': 1, 'xref': 159, 'page': 6, 'to': Point(613.6, 365.208), 'zoom': 0.0, 'collapse': True}]
[3, '2.1 ETAT DES LIEUX INTERNATIONAL', 7, {'kind': 1, 'xref': 161, 'page': 6, 'to': Point(622.214, 400.965), 'zoom': 0.0}]
[3, '2.2 ETAT DES LIEUX EUROPEEN', 8, {'kind': 1, 'xref': 166, 'page': 7, 'to': Point(86.2342, 134.505), 'zoom': 0.0}]
[3, '2.3 ETAT DES LIEUX FRANÇAIS', 9, {'kind': 1, 'xref': 164, 'page': 8, 'to': Point(86.2342, 63.4051), 'zoom': 0.0}]
[3, '2.4 ETAT DES LIEUX REGIONAL', 9, {'kind': 1, 'xref': 162, 'page': 8, 'to': Point(622.214, 394.005), 'zoom': 0.0}]

[1, 'CHAPITRE I : PREAMBULE', -1, {'kind': 0, 'xref': 128, 'page': -1, 'collapse': True, 'zoom': 0.0}]
[2, '1 CONTEXTE POLITIQUE ET ENGAGEMENTS', 7, {'kind': 1, 'xref': 129, 'page': 6, 'to': Point(1127.17, 77.6199), 'zoom': 0.0, 'collapse': True}]
[3, '1.1 CONTEXTE POLITIQUE INTERNATIONAL', 7, {'kind': 1, 'xref': 130, 'page': 6, 'to': Point(1009.51, 86.2342), 'zoom': 0.0}]
[3, '1.2 CONTEXTE POLITIQUE EUROPEEN', 7, {'kind': 1, 'xref': 131, 'page': 6, 'to': Point(782.595, 86.2342), 'zoom': 0.0}]
[3, '1.3 CONTEXTE POLITIQUE FRANÇAIS', 7, {'kind': 1, 'xref': 132, 'page': 6, 'to': Point(631.095, 86.2342), 'zoom': 0.0}]
[2, '2 ETAT DES LIEUX', 7, {'kind': 1, 'xref': 133, 'page': 6, 'to': Point(825.792, 613.6), 'zoom': 0.0, 'collapse': True}]
[3, '2.1 ETAT DES LIEUX INTERNATIONAL', 7, {'kind': 1, 'xref': 134, 'page': 6, 'to': Point(790.035, 622.214), 'zoom': 0.0}]
[3, '2.2 ETAT DES LIEUX EUROPEEN', 8, {'kind': 1, 'xref': 135, 'page': 7, 'to': Point(1056.49, 86.2342), 'zoom': 0.0}]
[3, '2.3 ETAT DES LIEUX FRANÇAIS', 9, {'kind': 1, 'xref': 136, 'page': 8, 'to': Point(1127.59, 86.2342), 'zoom': 0.0}]
[3, '2.4 ETAT DES LIEUX REGIONAL', 9, {'kind': 1, 'xref': 137, 'page': 8, 'to': Point(796.995, 622.214), 'zoom': 0.0}]
```

## Your configuration (mandatory)
 - Operating system, potentially version and bitness
 - Python version, bitness: 3.11
 - PyMuPDF version, installation method (**wheel** or **generated** from source).: 1.23.2
[landscape_rot.pdf](https://github.com/pymupdf/PyMuPDF/files/12476644/landscape_rot.pdf)

**Comment by JorjMcKie** [comment]
Thanks for reporting this.
Will have a look.

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.23.8.
